rust/core-lib/src/tabs.rs

```diff
@@ -567,10 +568,19 @@ impl CoreState {
     fn finalize_new_views(&mut self) {
         let to_start = mem::replace(&mut self.pending_views, Vec::new());
         to_start.iter().for_each(|(id, config)| {
-            let modified = self.detect_whitespace(*id, config);
-            let config = modified.as_ref().unwrap_or(config);
-            let mut edit_ctx = self.make_context(*id).unwrap();
-            edit_ctx.finish_init(&config);
+            let buffer_id = self.views.get(&id).map(|v| v.borrow().get_buffer_id()).unwrap();
```

Comment:
The `detect_whitespace` method already fetches the `buffer_id` - if you make the check over there, you can just return `None` on false and keep `finalize_new_views` simpler?

Reply:
That's a good point, this becomes basically a one-liner in `detect_whitespace`: 
```rust
if editor.borrow().get_buffer().is_empty() { return None; }
```

Reply:
Thanks for your advice!